[quant][graphmode] Dynamic quant: Insert observers for module output

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39458 [quant][graphmode] Dynamic quant: Insert observers for module output**
* #39412 [quant][graphmode] Dynamic Quant : Do not depend on input shapes

Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21872939](https://our.internmc.facebook.com/intern/diff/D21872939)

supriyar committed:
[quant][graphmode] Dynamic quant: Insert observers for module output
Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/passes/quantization/insert_observers.cpp:
```diff
@@ -1063,25 +1063,19 @@ InsertObserversHelper::insertObserversFor(
-              recordObserved(
-                  v, *observer_opt, values_to_observe, block_observed_values);
-            }
+      }
```
I just found that we can use isObserved in L1054-1055, could you change that?

jerryzh168 reviewed (approved):
LG

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21872939](https://our.internmc.facebook.com/intern/diff/D21872939)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21872939](https://our.internmc.facebook.com/intern/diff/D21872939)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21872939](https://our.internmc.facebook.com/intern/diff/D21872939)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21872939](https://our.internmc.facebook.com/intern/diff/D21872939)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21872939](https://our.internmc.facebook.com/intern/diff/D21872939)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21872939](https://our.internmc.facebook.com/intern/diff/D21872939)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21872939](https://our.internmc.facebook.com/intern/diff/D21872939)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Dynamic quant: Insert observers for module output"


Summary:
Previously if we had a CallMethod followed by a CallFunction, we didn't check for observers at output of CallMethod since it was handled separately.
This change makes it default to check the outputs of all nodes to identify values that need observers

Test Plan:
python test/test_quantization.py test_dynamic_shared_weights

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21872939](https://our.internmc.facebook.com/intern/diff/D21872939)

[ghstack-poisoned]
